### PR TITLE
Test pagination code

### PR DESCRIPTION
**Pull request by jabielecki**
Test pagination code without concurrency.

Test pagination code with concurrency (GETs do not happen when POST/PUT/DELETE is running, and vice versa).

No changes to functionality except comments.

### CODE REVIEW

File: client_pages_test.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		Reply(200).
+		BodyString(`{"response":[4],"extra":"x"}`)
+
+	res, err := client.Get("/url")
+	assert.NoError(t, err)
+	assert.Equal(t, `{"response":[1,2,3,4],"extra":"x"}`, res.Raw)
+}
+
+// TestClientGet_FirstNonArray tests the Client.Get against non-array attribute "response".
+func TestClientGet_FirstNonArray(t *testing.T) {
```
Comment (danischm) [inline client_pages_test.go]: Isn't this already covered by the GET test in client_test.go? If we want to verify the response body content, we could maybe add that to the existing test.

Comment (jabielecki) [reply client_pages_test.go]: done, thanks

File: client_test.go [hunk 2]
```diff
@@ -70,6 +70,10 @@ func TestClientGet(t *testing.T) {
 	_, err = client.Get("/url")
 	assert.NoError(t, err)
 
+	// Attribute "response" can be non-array - it doesn't cause pagination failure.
+	gock.New(testURL).Get("/url").Reply(200).BodyString(`{"response":"a string"}`)
+	assert.NoError(t, err)
```
Comment (danischm) [inline client_test.go:71]: We are missing the `client.Get()` call here, though I think we can add the Body mock to the previous test at line #69 and then add this check after line #71:
```
assert.Equal(t, "a string", res.Get("response").String())
```

**Review by danischm — CHANGES_REQUESTED** [review]

File: client_test.go (hunk 2 above)
Comment (jabielecki) [reply client_test.go:71]: Oh, what, terribly sorry - fixed it as you suggested.

**Review by danischm — APPROVED** [review]

**Comment by danischm** [comment]
Thx!

**Pull request merged**